p2panda-net/src/engine/gossip.rs

```diff
@@ -205,13 +207,17 @@ where
     async fn on_joined(&mut self, topic_id: [u8; 32], stream: GossipTopic) -> Result<()> {
         self.joined.insert(topic_id);
 
-        // Split the gossip stream and insert handles to the receiver and sender
+        // Split the gossip stream and insert handles to the receiver and sender.
         let (stream_tx, stream_rx) = stream.split();
+
+        // Collect all our current direct neighbors for this gossip topic.
+        let peers: Vec<PublicKey> = stream_rx.neighbors().collect();
```

Comment:
This is how we learn who we have just connected to (ie. the peer / peers who served as our entry-point into the gossip overlay).